File: 0169-majority-element/0169-majority-element.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        int ele=nums[0];
        int n=nums.size();
        int cnt=1;
        for(int i=1;i<n;i++){
           if(cnt==0){
            ele=nums[i];
           }
           if(nums[i]==ele){
            cnt++;
           }else{
            cnt--;
           }
        }
        return ele;
    }
};

File: 0169-majority-element/0169-majority-element.java
class Solution {
    public int majorityElement(int[] nums) {
        Map<Integer,Integer>mp=new HashMap<>();
        int n=nums.length;
        for(int i=0;i<n;i++){
            mp.put(nums[i],mp.getOrDefault(nums[i],0)+1);
        }
        for(Integer nm:mp.keySet()){
            if(mp.get(nm)>n/2){
                return nm;
            }
        }
        return -1;
    }
}